mitodl/course-search-utils
# add useSearchQueryParams and useInfiniteSearch

### What are the relevant tickets?
#65 

### Description (What does it do?)
The goal of this PR is to make this library easier for consuming apps to use and simplify the implementation of some of the hooks. I add two hooks:
1. `useSearchQueryParams`: derive search API parameters from the URLSearchParams and return functions to help update URLSearchParams
2. `useInfiniteSearch`: make paginated search API calls use search params returned by previous hook. 

### How can this be tested?
 To see the hooks in use (outside of tests), I made a branch in MIT Open using the hooks in this PR https://github.com/mitodl/mit-open/tree/cc/search-page-experiment. (It doesn't directly use this branch, it uses this branch cherry-picked to #69).

In MIT Open repo:
1. `git fetch`, `git checkout cc/search-page-experiment`. 
2. View `SearchPage.tsx` code
3. View http://localhost:8063/search/ and try out the page: filtering facets, scrolling, etc.
    - facet state should update
    - URL should update
    - search results should be accurate
    - inspect API calls in network tab 


https://github.com/mitodl/course-search-utils/assets/9010790/35bc43e9-d90d-4df1-8b6b-68b6c127d96d
